share/scripts/sdcdb.tcl

```diff
@@ -729,16 +729,22 @@ proc print_info {} {
     if {$file eq {}} {
         error "address [h [reg PC]] not found in database."
     }
-    print_status $file $line
+    puts -nonewline [check_status $file $line]
 }
 
-proc sdcdb_info {arg} {
-    if {$arg eq "-break"} {
+proc sdcdb_info {{param {}}} {
+    if {$param eq "-break"} {
         debug break
     }
-    if {[catch {print_info} fid]} {
-        puts $fid
+    set result {}
+    if {[catch {set result [check_info]} msg]} {
+        variable debug
+        if {$debug} {
+            debug_out stderr $::errorInfo
```

Comment:
The check for `debug` is already part of `debug_out`.

(Maybe something to check (I'm not 100% sure it's true): `variable debug` should _only_ be used in the proc `debug_out`).